en/components/general/update-guide.md

```diff
@@ -41,6 +41,172 @@ Unfortunately not all changes can be automatically updated. Changes below are sp
+        ```html
+        <igx-grid customDirective ...></igx-grid>
+        ```
+
+        ```typescript
+        @Directive({
+            selector: '[customDirective]'
+        })
+        export class customDirective {
+
+        constructor(@Host() @Optional() @Inject(IGX_GRID_BASE) grid: IgxGridBaseDirective) { }
+        ```
+- `RowDirective`, `RowType`
+    - **Breaking Change** - `rowData` and `rowID` properties are removed from `RowDirective` and from classes implementing the `RowType` interface. Use `data` and `key` instead. Use `ng update` for automatic migration. Automatic migration will not be able to pick up some examples from templates, where the template context object is not typed:
+        ```html
+        <ng-template igxCell let-cell="cell">
+            <span>{{ cell.rowID }}</span>
+            <span>{{ cell.row.rowData.ProductID }}</span>
+        </ng-template>
+        ```
+        Update such templates manually to
+        ```html
+        <span>{{ cell.key }}</span>
+        <span>{{ cell.row.data.ProductID }}</span>
+        ```
+- `igxGrid`
+    - Exposed a `groupStrategy` input that functions similarly to `sortStrategy`, allowing customization of the grouping behavior of the grid. 
+- `IgxCsvExporterService`, `IgxExcelExporterService`
+    - Exporter services are no longer required to be provided in the application since they are now injected on a root level.
+- `IgxGridToolbarPinningComponent`, `IgxGridToolbarHidingComponent`
+    - Exposed new input `buttonText` which sets the text that is displayed inside the dropdown button in the toolbar.
+- `IgxCombo`
+    - Added `groupSortingDirection` input, which allows you to set groups sorting order.
+- `IgxGrid`, `IgxTreeGrid`, `IgxHierarchicalGrid`
+    - Added new directives for re-templating header sorting indicators - `IgxSortHeaderIconDirective`, `IgxSortAscendingHeaderIconDirective` and `IgxSortDescendingHeaderIconDirective`.
+- `IgxDialog`
+    - Added `focusTrap` input to set whether the Tab key focus is trapped within the dialog when opened. Defaults to `true`.
+- `IgxColumnActionsComponent`
+    - **Breaking Change** - The following input has been removed
+        - Input `columns`. Use `igxGrid` `columns` input instead.
+- `IgxCarousel`
+    - **Breaking Changes** -The carousel animation type `CarouselAnimationType` is renamed to `HorizontalAnimationType`.
+- `IgxGridStateDirective` - now supports `disableHiding` column prop and column groups
+
+### Theming
+* Sass Modules:
+The theming engine has switched to [Sass modules](https://sass-lang.com/documentation/at-rules/use). This change means all theming library functions(comopnent themes, etc.), mixins(component mixins, etc.), and variables are now being `forwarded` from a single file. To correctly use the Sass theming library, your project should utilize Dart Sass version 1.33.0 or later and change all imports of the theming library from:
+
+```scss
+@import '~igniteui-angular/lib/core/styles/themes/index';
```

Comment:
Give an example with both the trial and the licensed packages in the code snippets.